src/InfoViz/Core/MutualInformationProvider/index.js

```diff
@@ -39,7 +39,9 @@ function mutualInformationProvider(publicAPI, model) {
   };
 
   function updateHistogram2D(histograms) {
-    if (Object.keys(histograms).length > 1) {
+    // even if histograms only has maxCount = 0, run through PMI.updateMI
+    // so deltaHandling.removed array is handled properly.
+    if (Object.keys(histograms).length >= 1) {
```

Comment:
Absolutely right. Moved the check around the loop which will do nothing with length = 1.